bokehjs/src/coffee/source/ajax_data_source.coffee

```diff
@@ -16,20 +16,27 @@ define [
 
     setup : (plot_view, glyph) =>
       @pv = plot_view
-      @update()
+      @get_data(true)
       if @get('polling_interval')
-        @interval = setInterval(@update, @get('polling_interval'))
+        @interval = setInterval( @get_data, @get('polling_interval'), @get('overwrite'), @get('max_size'))
 
-    update : () =>
+    get_data : (overwrite, max_size=0) =>
       $.ajax(
         dataType: 'json'
+        ifModified: true
         url : @get('data_url')
         xhrField :
           withCredentials : true
         method : @get('method')
         contentType : 'application/json'
       ).done((data) =>
-        @set('data', data)
+        if overwrite == true
+          @set('data', data)
+        else
+          original_data = @get('data')
+          data.x = original_data.x.concat(data.x)[-max_size..]
+          data.y = original_data.y.concat(data.y)[-max_size..]
+          @set('data', data)
```

Comment:
there should be a clever way than recreating the whole dataset by just pushing values on top of that actual dataset?